server/nimble_server.py

```diff
@@ -49,33 +68,47 @@ async def get_body(request: Request):
     req = await request.json()
     config = req['config']
 
+    # Make sure that a config was passed
+    if (len(config) == 0):
+        print("Nothing to build")
+        return ORJSONResponse([{"error": "Configuration must have components in it."}], status_code=500)
+
     print("Starting build for config:")
     print(config)
 
-    # Generate a hash of the configuration here to identify it
-    # sort config by key, get "key=value" strings, concatenate
-    config_hash = "+".join([f"{key}={config[key]}" for key in sorted(config.keys())])
-    # delete all non-alphanumeric characters (other than _+=)
-    config_hash = re.sub(r'[^a-zA-Z0-9_+=]', '', config_hash)
-    
-    print("config_hash is: " + config_hash)
+    # Create a clean, unique name for this build
+    unique_name = get_unique_name(config)
 
-    # If the build has been performed before, send the cached file to speed the system up
+    # Trigger the generation of all materials, but only if they do not already exist
     module_path = Path(__file__).resolve().parent.parent
-    build_path = module_path  / "build" / config_hash
-    if not os.path.exists(str(build_path) + ".zip"):
-        # Trigger the orchestration script
-        generate_docs(config, config_hash)
+    if not os.path.exists(module_path / "_cache_" / f"{unique_name}.zip"):
+        # Call all the underlying Nimble generator code
+        generate(config)
 
-    # Make sure that there is a valid config before passing it on to the orchestration script
-    if config is not None and "server_1" in config.keys():
-        # Save this generated config so that it can be polled
-        generated_docs_urls[config_hash] = server_base_url + "/wakoma/nimble/auto-doc?config_hash=" + urllib.parse.quote(config_hash)
+        # Create the zip file
+        shutil.make_archive(str(module_path / "_cache_" / unique_name), 'zip', os.path.join(module_path, "build"))
 
-        return ORJSONResponse([{"redirect": generated_docs_urls[config_hash], "description": "Poll this URL until your documentation package is ready."}])
-    else:
-        return ORJSONResponse([{"error": "Configuration must be a valid JSON object."}])
+        # Make a copy of the glTF preview file to cache it
+        shutil.copyfile(str(module_path / "build" / "assembly" / "assembly.glb"), str(module_path / "_cache_" / f"{unique_name}.glb"))
+
+        # Make a cached copy of the assembly docs so that they can be served to the user
+        copy_tree(str(module_path / "build" / "assembly-docs"), str(module_path / "server" / "static" / "builds" / f"{unique_name}_assembly_docs"))
+
+        # Make sure the new directory is served as static
+        # app.mount(str(module_path / "_cache_" / f"{unique_name}-assembly-docs"), StaticFiles(directory="static"), name="static")
```

Comment:
Maybe remove commented out code or say why it is commented out?

Reply:
This commented code is a placeholder for how the dynamic static serving of the cached assembly-docs could/should? work. I'd like to leave it in for one more revision.

Reply:
Maybe we just add a comment to say that so it is clear?